server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/SystemMailboxesProviderImplTest.java

```diff
@@ -19,15 +19,141 @@
 
 package org.apache.james.jmap.utils;
 
-import org.junit.Ignore;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.when;
+
+import java.util.stream.Stream;
+
+import org.apache.james.jmap.exceptions.MailboxRoleNotFoundException;
+import org.apache.james.jmap.model.mailbox.Role;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.manager.MailboxManagerFixture;
+import org.apache.james.mailbox.mock.MockMailboxSession;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MailboxMetaData;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MailboxQuery;
+import org.apache.james.mailbox.model.TestId;
+import org.apache.james.mailbox.store.SimpleMailboxMetaData;
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import com.google.common.collect.ImmutableList;
 
 public class SystemMailboxesProviderImplTest {
 
-    @Ignore("1716 this class needs a test suite")
+    private static final MailboxPath INBOX = MailboxManagerFixture.MAILBOX_PATH1;
+    private static final MailboxPath OUTBOX = MailboxManagerFixture.MAILBOX_PATH2;
+    private static final char DELIMITER = '.';
+
+    private static MailboxId inboxId = TestId.of(1);
+    private static MailboxId outboxId = TestId.of(1);
+
+    private static MailboxMetaData inboxMetadata = new SimpleMailboxMetaData(INBOX, inboxId, DELIMITER);
+    private static MailboxMetaData outboxMetadata = new SimpleMailboxMetaData(OUTBOX, outboxId, DELIMITER);
+
+    private MailboxSession mailboxSession = new MockMailboxSession("user");
+    private SystemMailboxesProviderImpl systemMailboxProvider;
+
+    @Mock private MailboxManager mailboxManager;
+
+    @Mock private MessageManager inboxMessageManager;
+    @Mock private MessageManager outboxMessageManager;
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
```

Comment:
```
@RunWith(MockitoJUnitRunner.class)
```

On top of the class...

http://stackoverflow.com/questions/15494926/initialising-mock-objects-mockito

This will make this call avoidable :-)